### PR TITLE
Week 5 rashad gilliam

### PR DESCRIPTION
**Pull request by rgilliam83**
# Submitting Week 5 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [x] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

**Comment by WalkerTexas-coder** [comment]
Anything I can do to help out with getting the assessment in on time? Make sure to communicate ahead of time if you are not going to get your work in when it's expected.

File: code-challenges.test.js [hunk 1]
```diff
@@ -14,39 +19,297 @@
...
+//establish parameters for arr
+//generate for loop
+//set conditionals to return expected boolean response
+
+
+//To check full house
+const fullHouse = (arr) => {
+    let countObj = {}
+    for(let x of arr){
+      countObj[x] = (countObj[x] || 0) + 1;
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js]: Good job!

File: code-challenges.test.js [hunk 2]
```diff
@@ -14,39 +19,297 @@
...
+// info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
+
+
 
 // b) Create the function that makes the test pass.
+
+//create an object which holds the count of each of the value.
+//establish parameters for arr
+//generate for loop
+//set conditionals to return expected boolean response
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js]: I would like to see this pseudocode describe the for loop working with the object

File: interview-questions.md [hunk 3]
```diff
@@ -8,44 +8,45 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 1. What is a hash in Ruby?
 
-Your answer:
+Your answer:  A Hash is a collection of unique keys and their values. Also called associative arrays, they are similar to Arrays, but where an Array uses integers as its index, a Hash allows you to use any object type. Hashes enumerate their values in the order that the corresponding keys were inserted. Hashes allow an alternate syntax for keys that are symbols. Hashes have a default value that is returned when accessing keys that do not exist in the hash. If no default is set nil is used.
 
-Researched answer:
+Researched answer:  The order in which you traverse a hash by either key or value may seem arbitrary and will generally not be in the insertion order. If you attempt to access a hash with a key that does not exist, the method will return nil. In Ruby, Hash is a set of keyt value pairs. With a hash, the indexing is done with the help of keys of any object type. In Hash, the order of returning keys and their value by various iterators is arbitrary and will generally not be in the insertion order. The default value of Hashes is nil. When a user tries to access the keys which do not exist in the hash, then the nil value is returned.
```
Comment (WalkerTexas-coder) [inline interview-questions.md]: [Found the article you copied this from. ](https://www.tutorialspoint.com/ruby/ruby_hashes.htm#:~:text=The%20order%20in%20which%20you%20traverse%20a%20hash%20by%20either%20key%20or%20value%20may%20seem%20arbitrary%20and%20will%20generally%20not%20be%20in%20the%20insertion%20order.%20If%20you%20attempt%20to%20access%20a%20hash%20with%20a%20key%20that%20does%20not%20exist%2C%20the%20method%20will%20return%20nil.)

**Review by WalkerTexas-coder — CHANGES_REQUESTED** [review]
Learn has a policy that if an Interview answer is found to be copied in part or in whole that all of the interview answers will need to be re submitted in your own words before this Assessment can be accepted and you can move forward with the next assessments. 

I have found the article you copied your first answer from. Please re submit all of the interview questions and respective answers so that I can finish reviewing your assessment. Please refer to the instructions at the top of the interview questions file for the expectations.

File: code-challenges.test.js (hunk 2 above)
Comment (rgilliam83) [reply code-challenges.test.js]: understood

File: interview-questions.md (hunk 3 above)
Comment (rgilliam83) [reply interview-questions.md]: understood

**Review by WalkerTexas-coder — APPROVED** [review]
Great work!

**Pull request merged**